cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/planner/PipelinePlanner.java

```diff
@@ -382,12 +383,48 @@ private PipelinePhase dagToPipeline(Dag dag, Map<String, String> connectors,
 
       // add other plugin types
       StageSpec spec = specs.get(stageName);
-      phaseBuilder.addStage(spec);
+      phaseBuilder.addStage(updateStageSpec(spec, conditionConnectors));
     }
-
     return phaseBuilder.build();
   }
 
+  private StageSpec updateStageSpec(StageSpec spec, Map<String, String> conditionConnectors) {
```

Comment:
nit: rename to something like `fixConditionConnectorNames`

It would be better if we didn't need to correct this at all and just created it correctly the first time. Is that not possible because we need the entire Map of conditionConnectors in order to do that?

Reply:
can you comment on why we can't create the correct StageSpec to begin with?

Reply:
ok I think I understand. I think it's ok to keep it in the planner where you have it, since that's also the place where it is converting the connector stages into corresponding sources/sinks.